packages/cli/src/commands/account/claim-domain.ts

```diff
@@ -21,5 +21,10 @@ export default class ClaimDomain extends ClaimCommand {
     const metadata = this.readMetadata()
     await this.addClaim(metadata, createDomainClaim(res.flags.domain))
     this.writeMetadata(metadata)
+
+    const signature = JSON.parse(metadata.toString()).meta.signature
```

Comment:
I realized that my original write-up of the issue was incorrect. I don't think we should be putting the signature of the whole metadata in the TXT record, but just the hash of the claim itself, that avoids the need to have to actually pull the metadata, and one can verify the claim independently (assuming one verifies that the signature of the metadata proves the integrity of the claim itself)

Reply:
As you said, I think makes sense to use the signature of claim itself and not the whole metadata. At the same time I think is useful to have the mechanism of given a domain and the metadata url:
1. download the metadata
2. find if there is a claim inside the metadata referenced in the url provided by the user
3. hash the claim found in the metadata

In that case you not only validate the user owns the domain (via TXT record), also indirectly validate that Metadata URL is valid as a user claims catalog.

While this part is discussed, I will change the code to use as signature the domain claim instead the whole metadata

Reply:
I agree that the whole metadata should be verified. However I would leave that to the caller. The primary means for users to verify is via `get-metadata` anyways which already verifies every single claim, right?

Reply:
yep, this is using the same plumbing existing in the `IdentityMetadataWrapper` so now there is no much custom implementation about that